src/utils/image-processor.js

```diff
@@ -274,29 +273,121 @@ const replaceJpegChunk = function (blob, marker, chunks) {
 }
 
 const getExif = function (file) {
-  var exif, op
-  exif = null
-  op = readJpegChunks(file)
+  let exif = null
+  const op = readJpegChunks(file)
   op.progress(function (pos, l, marker, view) {
     if (!exif && marker === 0xe1) {
       if (view.byteLength >= 14) {
-        if (view.getUint32(0) === 0x45786966 && view.getUint16(4) === 0) {
+        if (
+          // check for "Exif\0"
+          view.getUint32(0) === 0x45786966 &&
+          view.getUint16(4) === 0
+        ) {
           exif = view
           return exif
         }
       }
     }
   })
   return op.then(
-    function () {
-      return exif
-    },
-    function (reason) {
-      return $.Deferred().reject(exif, reason)
+    () => exif,
+    () => $.Deferred().reject(exif)
```

Comment:
I've removed returning of `reason` because it isn't used anywhere and it changes the number of arguments [here](https://github.com/uploadcare/uploadcare-widget/pull/902/files#diff-efb725f4c993c9bdf24c09ea9dffa35a088d66af389523ec61cdd7cb01cf78dbR48)